src/common/abstractPromote.ts

```diff
@@ -77,18 +85,26 @@ export abstract class PromoteCommand<T extends typeof SfCommand> extends SfComma
     this.sourceStageId = this.getSourceStageId();
     const asyncOperationId: string = await this.requestPromotion(doceOrg);
 
-    // TODO: move this to logger service
-    this.log(`Job ID: ${asyncOperationId}`);
-    await DeployPipelineCache.set(asyncOperationId, {});
+    this.outputService.setAorId(asyncOperationId);
+
+    this.outputService.printAorId();
+    await this.outputService.printOpSummary();
 
     if (this.flags.async) {
-      // TODO display async message
+      await DeployPipelineCache.set(asyncOperationId, {});
```

Comment:
I think this was moved outside the if statement because we always need to set the cache.
Do you remember this @EstebanRomero84 ?

Reply:
I remember why now !!
If we want to resume an operation that timeout, we need to have that aorid on the cache to resume it using the most recent

Reply:
Yes, you are right.